### DISCUSSION TITLE
How to resolve WP Post Object by custom mutation ?

### DISCUSSION THREAD

**Discussion by jiwon-mun** [wp-graphql/wp-graphql#2702] [Q&A] [answered]
```php

add_action( 'graphql_register_types', function() { 

	register_graphql_mutation( 'viewPopularPost', [

		'inputFields'         => [
			'postId' => [
				'type' => 'Int',
				'description' => __( 'input field' ),
			]
		],
	
		'outputFields'        => [
			'post' => [
				'type' => 'Post',
				'description' => __( 'output field' ),
				'resolve' => function( $payload, $args, $context, $info ) {
								graphql_debug($payload['post']);// I can check I got Post here!
							   return isset( $payload['post'] ) ? $payload['post'] : null;
				}
			]
		],
	

		'mutateAndGetPayload' => function( $input, $context, $info ) {
			$post = get_post($input['postId']);
			return [
				'post' => $post,
			];
		}
	] );

});
```

I wrote like this, but empty return resolve.

<img width="768" alt="image" src="https://user-images.githubusercontent.com/109935474/215337294-f777433b-c555-42cd-97b7-6e3005eca7fb.png">

i checked by `graphql_debug`, and I could know the `$payload` has correct post object.

but why did i get a something like empty object, and how to resolve WP Post Object.

Please help me 😭

**Comment by justlevine** [comment, marked as answer]
Hey @jiwon-mun ,

That's because you need to return a [WPGraphQL `Post` model](https://github.com/wp-graphql/wp-graphql/blob/f68ed725a753c91aabeaa59aca61d3a1f3d51ba7/src/Model/Post.php#L116) and not the `WP_Post` object.

Changing your return statement in `$config['outputFields']['post']['resolve']` to the following should fix your issue:

```diff
- return isset( $payload['post'] ) ? $payload['post'] : null;
+ return isset( $payload['post'] ) ? new \WPGraphQL\Model\Post( $payload['post'] ) : null;
```

**Reply by jiwon-mun** [reply]
It works! Thank you :) 🤩
